contrib/golang/filters/http/test/golang_integration_test.cc

```diff
@@ -429,6 +492,28 @@ name: golang
     }
   }
 
+  void testDynamicMetadata(std::string path) {
+    initializeBasicFilter(BASIC, "*", true);
+
+    codec_client_ = makeHttpConnection(makeClientConnection(lookupPort("http")));
+    Http::TestRequestHeaderMapImpl request_headers{
+        {":method", "POST"},        {":path", path},
+        {":scheme", "http"},        {":authority", "test.com"},
+        {"x-set-metadata", "true"},
+    };
+
+    auto encoder_decoder = codec_client_->startRequest(request_headers);
+    Http::RequestEncoder& request_encoder = encoder_decoder.first;
+    auto response = std::move(encoder_decoder.second);
+    codec_client_->sendData(request_encoder, "helloworld", true);
+
+    waitForNextUpstreamRequest();
+    // check no dynamic metadata
+    EXPECT_EQ(true,
+              upstream_request_->headers().get(Http::LowerCaseString("dy-envoy-lb-foo")).empty());
```

Comment:
this looks a bit weired to me, there is no where set the `dy-envoy-lb-foo` header in this PR.

Reply:
fixed in 553978d77afb290bbded504d86e2d6ddf4844f58